components/p3a/message_manager.cc

```diff
@@ -146,6 +149,16 @@ void MessageManager::UpdateMetricValue(
   if ((update_for_all || !*only_update_for_constellation) && json_log_store) {
     json_log_store->UpdateValue(std::string(histogram_name), bucket);
   }
+  const auto* metric_config = GetMetricConfig(histogram_name);
+  if (metric_config && *metric_config &&
+      (*metric_config)->record_activation_date) {
+    // Record activation date for metric, for retention measurement purposes
+    ScopedDictPrefUpdate update(&*local_state_, kActivationDatesDictPref);
+    if (!update->contains(histogram_name)) {
+      update->Set(histogram_name,
+                  static_cast<double>(base::Time::Now().ToTimeT()));
```

Comment:
base::TimeToValue?

Reply:
i keep forgetting about this function :facepalm: